### ISSUE TITLE
Print more explicit info when requesting package that is also pinned

### ISSUE THREAD

**Issue by buhrmann** [mamba-org/mamba#735]
Hi, it seems in some cases mamba isn't any more forthcoming about conflicts then conda. E.g. in a particular environment, when I try to install pynndescent=0.4.7 (from conda-forge):

``` bash
  Package      Version  Build           Channel                    Size
─────────────────────────────────────────────────────────────────────────
  Install:
─────────────────────────────────────────────────────────────────────────

  libllvm8       8.0.1  hc9558a2_0      conda-forge/linux-64     Cached
  llvmlite      0.30.0  py37h8b12597_1  conda-forge/linux-64     Cached
  numba         0.46.0  py37hb3f55d8_1  conda-forge/linux-64     Cached
  pynndescent    0.4.7  pyh8c360ce_0    conda-forge/noarch       Cached

  Summary:

  Install: 4 packages
```

But then trying to install pynndescent=0.4.8:

```
Looking for: ['pynndescent=0.4.8']

conda-forge/linux-64     Using cache
conda-forge/noarch       Using cache
pkgs/main/linux-64       Using cache
pkgs/main/noarch         Using cache
pkgs/r/linux-64          Using cache
pkgs/r/noarch            Using cache
Encountered problems while solving.
Problem: conflicting requests
```

And I can't seem to find any way in mamba to understand what has failed exactly? The repoquery e.g. returns the same dependencies in both cases, when I run `repoquery depends -c conda-forge pynndescent=0.4.7/8`:

```
 Name         Version Build              Channel             
──────────────────────────────────────────────────────────────
 pynndescent  0.4.8   pyh9f0ad1d_0       conda-forge/noarch  
 setuptools   23.0.0  py27_0             conda-forge/linux-64
 python       3.9.2   hffdb5ce_0_cpython conda-forge/linux-64
 numpy        1.20.1  py39hdbf815f_0     conda-forge/linux-64
 scipy        1.6.0   py39hee8e79c_0     conda-forge/linux-64
 scikit-learn 0.24.1  py39h4dfa638_0     conda-forge/linux-64
 llvmlite     0.30.0  py38h8b12597_1     conda-forge/linux-64
 numba        0.46.0  py37hb3f55d8_1     conda-forge/linux-64

```

(apart from the version of pynndescent itself of course).

Not sure if there is a problem with the package itself, or if there is any way to find out what's going wrong?

Incidentally, the versions shown in `repoquery depends` cannot be the (minimal) requirements of the package, right? In the example above e.g. python is always specified as 3.9.2, while the package also works with 3.7 e.g.

**Comment by wolfv** [comment]
hmm, yeah, it's not the ideal error message, for sure. I haven't seen this one, actually.

Would be interesting to see a conda list output of your environment (creating a fresh env with the version you indicated seems to work for me).

Are you pinning packages by any chance? Maybe printing `conda info` could give some further clues.

Also if you really want to debug this, then run mamba with `-vvv` and you will get _a lot_ of output that might lead you to the problem (where libsolv fails). If you find something interesting I'd be very curious!

At some point we should do some research into how to get the best error messages out of libsolv. There are quite a few "unknowns" though ...

**Comment by buhrmann** [comment]
Yes, the problem seems to have been because of a conflicting spec in the pinned file (which I think until recently mamba ignored?). I don't know how feasible an error message to that effect would be? I think conda outputs something like "Requested spec conflicts with pinned spec" or similar.

As for `repoquery depends`, the output of  `conda search -i some_pkg` seems a bit more helpful, e.g. for the same package as above (pynndescent=0.4.8) I get the following, which correctly lists the minimal dependencies of the package (only showing a single build number):

```
pynndescent 0.4.8 py_1
----------------------
file name   : pynndescent-0.4.8-py_1.conda
name        : pynndescent
version     : 0.4.8
build       : py_1
build number: 1
size        : 35 KB
license     : BSD-2-Clause
subdir      : noarch
url         : https://repo.anaconda.com/pkgs/main/noarch/pynndescent-0.4.8-py_1.conda
md5         : f5af783a9fca752decbe428aa6cb8658
timestamp   : 2020-07-08 01:25:10 UTC
dependencies: 
  - llvmlite >=0.30
  - numba >=0.46
  - numpy >=1.13
  - python >=3.6
  - scikit-learn >=0.19
  - scipy >=1.0
  - setuptools
```

Could it be that `mamba depends` always shows the newest compatible version, and conda the minimal one?

**Comment by wolfv** [comment]
@buhrmann it would be nice if you could elaborate which package was pinned so we can easily recreate the issue.

You can still use `conda search` from mamba (`mamba search`) so we support the same feature, technically, hehe. 

But I agree, the depends output is not great (there is an open issue https://github.com/mamba-org/mamba/issues/618 ).

Any help on these topics is highly appreciated!

**Comment by buhrmann** [comment]
In this case it was in fact the very same package that I was trying to install that was pinned to an earlier version hahaha. I must have ended up in this state after changing the version in one part of a "multi-stage" build, where each stage installs its own requirements. An earlier stage must have installed (or at least pinned) a newer version, and a later stage an older version.

**Comment by wolfv** [comment]
We now print all pinned packages in an environment so the messaging should be a bit clearer. Implemented in #775
